synapse/handlers/pagination.py

```diff
@@ -335,20 +335,18 @@ async def get_messages(
         user_id = requester.user.to_string()
 
         if pagin_config.from_token:
-            room_token = pagin_config.from_token.room_key
+            from_token = pagin_config.from_token
         else:
-            pagin_config.from_token = (
-                self.hs.get_event_sources().get_current_token_for_pagination()
-            )
-            room_token = pagin_config.from_token.room_key
+            from_token = self.hs.get_event_sources().get_current_token_for_pagination()
 
-        room_token = RoomStreamToken.parse(room_token)
+        if pagin_config.limit is None:
+            # This shouldn't happen as we've set a default limit before this
+            # gets called.
+            raise Exception("limit not set")
 
-        pagin_config.from_token = pagin_config.from_token.copy_and_replace(
-            "room_key", str(room_token)
-        )
+        room_token = RoomStreamToken.parse(from_token.room_key)
 
-        source_config = pagin_config.get_source_config("room")
+        from_token = from_token.copy_and_replace("room_key", str(room_token))
```

Comment:
This looks to me like we're round-tripping the same value? What is this actually doing? (It could probably have a comment added.

Reply:
Errr, I'm pretty sure it was pointless before and is pointless now.